sycl-compress/include/sycl-compress/sycl-compress.h

```diff
@@ -0,0 +1,57 @@
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+#pragma once
+
+#include <memory>
+
+namespace sycl_compress {
```

Comment:
sycl-compress library seems to be not related to SYCL at all. It also duplicated already existing library LLVM Support. We should avoid logic duplication and re-use existing library.

For the inspiration, you can take a look at offload code compression for HIP compiler - https://github.com/intel/llvm/commit/124d0b787b5d1ff4aa06bbd61a98d3fc344d0cc6.

Reply:
Fixed. I now link LLVMSupport library with SYCL RT and `clang-offload-wrapper` to do (de)compression.

Reply:
I wouldn't outline this code into a separate subproject, but I would maybe preserve it as a thin wrapper around `zstd` even if it is a copy of files from `LLVMSupport`.

The reason for that is one less dependency which may cause problems. You can already see that we build LLVM differently from SYCL RT on Windows and can't link the two together.

We have online compiler extension which dynamically loads `ocloc` that is linked to LLVM. Well, even our OpenCL and L0 backends are linked to LLVM. I'm not saying that it will necessarily lead to problems, but it theoretically can. An example I've found is https://stackoverflow.com/questions/29283116/is-libgcc-s-so-linked-both-statically-and-dynamically-in-the-same-process-ok